### DISCUSSION TITLE
ROCM for GPU 7900 XTX

### DISCUSSION THREAD

**Discussion by lojo** [ROCm/ROCm#1891] [General]
Hi,

Is there any official announcement available, **if or when** the 7900 XTX will be supported? Or a timeline...?

Any information is appreciated!

- Artur

**Comment by Mushoz** [comment]
@saadrahim has commented here: https://github.com/RadeonOpenCompute/ROCm/discussions/1836#discussioncomment-4586574

That he will ask internally whether he is able to make any statement regarding this. It's probably best to keep an eye on that topic. Hoping to get an answer soon (Even if that answer is that they cannot disclose additional information!) so that I can make a decision on what to do with this 7900 XTX sitting in a box.

**Reply by xiaobo1025** [reply]
请问我要怎么运行rocm呢

**Comment by steyRNG** [comment]
Also waiting, even just on linux is fine. Windows would be more comfortable, but i'll take whatever i can get at this point. The newest Driver for Windows has SOME Capabilities mixed in already (https://www.amd.com/en/support/kb/release-notes/rn-rad-win-23-1-2-kb) but not actual ROCm, so no luck on CUDA enabled stuff yet.

**Comment by Nicthien** [comment]
Damn, I hope I didn't make a bad choice by buying this card, the various discussions about a Rocm and 7900XTX compatibility make me more and more afraid.

**Reply by steyRNG** [reply]
I'm also still wondering, using my old GPU in the meanwhile for ML Stuff, but not everyone has that option. Lets wait until the release of ROCM 5.5 / the next "officialy" supported GPU for ROCm and there will surely be Support for the GFX1100 Series / 7900 Series. But right now, we're just too early adopters.

**Comment by saadrahim** [comment]
RDNA3 support will be available officially in the fall this year.

https://community.amd.com/t5/rocm/new-rocm-5-6-release-brings-enhancements-and-optimizations-for/ba-p/614745/jump-to/first-unread-message

**Reply by sdli1995** [reply]
Dose RDNA3 will get it‘s claimed performance？In current rocm version 5.5.1 or 5.6 ，I run both train and inference successfully in 2* RX7900xtx but it  slower than 1 * rtx3090 in fp32 precision .

**Reply by johnnynunez** [reply]
@sdli1995 read the post: 
Fixed problems: We’ve also seen tremendous interest from developers wanting to run the ROCm open software platform for AI and ML on our Radeon™ consumer and Radeon™ Pro workstation GPUs and have heard the community challenges with specific driver issues on unsupported GPUs. I’m pleased to say that we have fixed the reported issues in ROCm 5.6 and we are committed to expanding our support going forward.  

Official support RDNA3: We plan to expand ROCm support from the currently supported[ AMD RDNA 2 workstation GPUs: the Radeon Pro v620 and w6800](https://docs.amd.com/en/docs-5.0.2/CHANGELOG.html#radeon-pro-v620-and-w6800-workstation-gpus) to select AMD RDNA 3 workstation and consumer GPUs. Formal support for RDNA 3-based GPUs on Linux is planned to begin rolling out this fall, starting with the 48GB Radeon PRO W7900 and the 24GB Radeon RX 7900 XTX, with additional cards and expanded capabilities to be released over time.

**Reply by sdli1995** [reply]
Another problem is suspend problem . After boot i need run rocm-smi loop to make video card driver work correct ,if i don't do this every time program call video card, dmesg shows driver reload log and warnnings

**Reply by saadrahim** [reply]
Have you filed a bug report?

**Comment by johnnynunez** [comment]
> RDNA3 support will be available officially in the fall this year.
> 
> https://community.amd.com/t5/rocm/new-rocm-5-6-release-brings-enhancements-and-optimizations-for/ba-p/614745/jump-to/first-unread-message

The next step is systolic arrays(matrix cores(CDNA)/tensor cores(Nvidia) on RDNA4 and future consoles. This increments the speed and reduce latency in  inference applications(FSR3) and a lot of type of models in games and AI. And if added to future PS5 Pro, it would be a masterful marketing move for AMD as an AI company.

**Comment by husmen** [comment]
I will believe it when I see it, we got excited before for nothing.

**Comment by fkam18** [comment]
Today's July 29, thought to buy a 7900 to run rocM on Linux, only to find none in AMD page, but surprisingly found support for Windows 🤣🤣 Thought rocM is featuring open, and Linux naturally is the de facto open but ending up Windows as a proprietary platform earns the fast lane. Might as well go with Nvidia if it's not for the openness 🤣😂

Logically speaking, enterprises already have Windows support and licensing and have pro grade GPUs budget while the individual developers are more on Linux with consumer grade GPUs hence it is more sensible to open up consumer grade Linux GPU support asap. AMD needs adoption to boost community in an open ecosystem I think

**Reply by johnnynunez** [reply]
> Today's July 29, thought to buy a 7900 to run rocM on Linux, only to find none in AMD page, but surprisingly found support for Windows 🤣🤣 Thought rocM is featuring open, and Linux naturally is the de facto open but ending up Windows as a proprietary platform earns the fast lane. Might as well go with Nvidia if it's not for the openness 🤣😂
> 
> 
> 
> Logically speaking, enterprises already have Windows support and licensing and have pro grade GPUs budget while the individual developers are more on Linux with consumer grade GPUs hence it is more sensible to open up consumer grade Linux GPU support asap. AMD needs adoption to boost community in an open ecosystem I think 

Windows support:
Rocm is available(HIP)
For AI: It's needed MiOpen(PR are open) and migraphax

**Comment by johnnynunez** [comment]
7900xtx available. I have 7900xt waiting for official support👌👌

**Reply by saadrahim** [reply]
It is now officially supported on Ubuntu 22.04

**Reply by johnnynunez** [reply]
when 7900xt and other? And when PR of windows will be accepted in MiOpen?

**Reply by saadrahim** [reply]
Apologies, I didn't notice the difference in your comment about 7900 XTX versus 7900 XT. My above comment is not correct.

**Comment by OmerFarukOruc** [comment]
Any tutorials for 7900 xtx to run rocm in Ubuntu?

**Comment by piotroxp** [comment]
I am planning to run AI on my Archlinux machine and plan to buy this card in Jan'24

Is support for this GPU working already? I really don't care if I go with NVIDIA, Im just an AMD fanboy but in case existential level productivity depends on it, Im happy to ditch the 7900

Anyone able to answer if we get actual f****** support for AMD gpus on linux?

**Reply by fluidnumericsJoe** [reply]
Radeon 7900 XTX is supported on Ubuntu 22.04 only. "Supported" means that AMD builds and tests ROCm distribution for a GPU on a given system. Since ROCm is open-source, you can attempt to build the packages you need from source code; the [spack package manager](https://spack.io) is quite useful for this purpose, though the install this route can take a while and may require some effort on your part.

Also, here's some [f****** documentation](https://rocm.docs.amd.com/en/latest/release/gpu_os_support.html) for you to check out regarding supported GPUs and Linux distributions.

**Reply by AnonymousRonin** [reply]
Buy nvidia.  And for the record, AMD's idea of support and mine must be vastly different.  If by support you mean it turns on, sure.  But  good luck getting gfx11 to load and stay stable.

**Reply by piotroxp** [reply]
We are thinking the same. I have been struggling to recompile tensorflow-rocm many times, and not once was I able to run anything on my 6600XT. With strong support missing, I believe AMD to be making one of the most stupid decisions in its history.

Its quite interesting that AMD, instead of just allowing all the GPUs out there to crush this AI boom with unleashing it for users is waiting for... well, really - what for? Profits bc of software toggles (AI card / non AI card?) ? 

Its really amusing that nVidia customers can just run tensorflow on Ubuntu and have fun, while AMD folks are getting this treatment.

I now own a Radeon, and the only reason I am considering the 7900 is due to actual support that was supposed to be made available by this EOY. Seems brand loyalty and taking care of customer productivity needs are not important for AMD, then.

**Reply by fluidnumericsJoe** [reply]
AMDs focus has primarily been on ensuring Frontier, Lumi, and Setonix (the three exascale HPC systems recently procured) were successful launches. 

This means that all of the ROCm efforts catered to the Instinct lineup of HPC grade cards. Now that that is done, it appears they're getting their game together on supporting consumer cards, but that process takes time. They still have to support instinct and now they need to hire and get folks out on the right teams to start properly supporting hardware solutions that are not on HPE Cray systems.

Not saying it's right or defending them in any way. It sounds like some context would be interesting for this discussion.

**Comment by johnnynunez** [comment]
For me it's working very well.
Use my scripts and build tensorflow and pytorch with your id card.
https://github.com/johnnynunez/rocm_lab
https://github.com/ROCmSoftwarePlatform/tensorflow-upstream/pull/2327

**Reply by AnonymousRonin** [reply]
On which distro?  Becuse contrary to popular belief, not everyone uses 1 of the 3 supported distros.  I still can't understand why they wouldn't support Debian and Arch directly. Everything I've tried has ultimately broke the distros I've tied them on, it even sent one into kernel panic...never seen that before.  That isn't support.  You would think by now that it ccould just  learn the env variables on it's own and configure it'self.  Guess that's still a ways out.

**Reply by johnnynunez** [reply]
> On which distro?  Becuse contrary to popular belief, not everyone uses 1 of the 3 supported distros.  I still can't understand why they wouldn't support Debian and Arch directly. Everything I've tried has ultimately broke the distros I've tied them on, it even sent one into kernel panic...never seen that before.  That isn't support.  You would think by now that it ccould just  learn the env variables on it's own and configure it'self.  Guess that's still a ways out.

Ubuntu 22.04.3

**Comment by AnonymousRonin** [comment]
I hope so because I can't stand Ubuntu. I use Kali for work, I've been using it since Backtrack4. Mind you this is only my desktop for remote work but it is bare metal and it's new in March. I have yet to get hashcat to run, I've completely lost everything twice and still no answers on this. So far my best option has been compiling llvm18 with clang and building out most of the llvm project. Minus the backends, that process just escapes me. I don't know what software Ubuntu runs to make it compatible but I almost gaurantee if I knew what it was I could configure my box to run it.

Sent from Proton Mail mobile

-------- Original Message --------
